proton-j/src/main/java/org/apache/qpid/proton/amqp/transport/ReceiverSettleMode.java

```diff
@@ -25,13 +25,32 @@
 
 import org.apache.qpid.proton.amqp.UnsignedByte;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public enum ReceiverSettleMode
 {
-    FIRST, SECOND;
+    FIRST(0),
+    SECOND(1);
+
+    private UnsignedByte value;
+    private static Map<UnsignedByte, ReceiverSettleMode> map = new HashMap<>();
+
+    private ReceiverSettleMode(int value) {
+        this.value = UnsignedByte.valueOf((byte)value);
+    }
 
-    public UnsignedByte getValue()
-    {
-        return UnsignedByte.valueOf((byte)ordinal());
+    static {
+        for (ReceiverSettleMode mode: ReceiverSettleMode.values()) {
+            map.put(mode.value, mode);
+        }
     }
 
+    public static ReceiverSettleMode valueOf(UnsignedByte value) {
+        return map.get(value);
```

Comment:
I think its worth throwing IAE if an out of bound value is provided, rather than just returning null.